provider/rancher/rancher.go

```diff
@@ -249,11 +249,24 @@ func (lbp *LBProvider) GetExistingConfigNames() (map[string]bool, error) {
 		return nil, fmt.Errorf("couldn't list lb services. Error: %#v", err)
 	}
 
-	if len(lbs.Data) == 0 {
+	var allLBs []client.LoadBalancerService
+	allLBs = append(allLBs, lbs.Data...)
+
+	isPartial := lbs.Pagination.Partial
+	for isPartial {
+		newlbs, err := lbs.Next()
+		if err != nil {
+			fmt.Printf("Failed to get Next set of lb services with error: %v", err)
```

Comment:
shall we return an error? Or if there is no next, error would be returned too? And if so, would it be possible to determine the diff between "no next" and "some other error"?

Reply:
When there is no next, there is no error thrown. It just sets the Pagination.Partial as false. 
We should be returning an error though